### ISSUE TITLE
cjxl ignores JPEG XMP if JPEG reconstruction is disabled

### ISSUE THREAD

**Issue by Hello71** [libjxl/libjxl#861]
SetBlobsFromJpegData reads exif and xmp. it is called via DecodeImageJPGCoefficients -> jpeg::DecodeImageJPG. in jxl::extras::DecodeImageJPG, exif is read by ReadExif. i guess there should be ReadXMP similar to ReadExif. it seems like a lot of code duplication though.

**Comment by jonsneyers** [comment]
Yes, it's a lot of unnecessary code duplication; I think it would make more sense to drop the libjpeg dependency and use our own jpeg parsing code for everything (currently it is used when jpeg reconstruction is enabled but not when it is disabled).

Ideally, imo, libjxl should be able to decode both jxl and jpeg to pixels — that would also be convenient for application developers, who can drop a dependency and a code path and conceptually just 'upgrade' libjpeg to libjxl instead of 'adding a new format'.

Once libjxl can do that, and cjxl is using libjxl, we can then allow jxl and jpeg input (as pixels) so you can also use cjxl to e.g. do lossy compression on a lossless jxl (instead of having to first decode it to png).

**Comment by moltenform** [comment]
Can confirm this occurs in 0.10.3, a jpg's xmp metadata is lost upon conversion to jxl when the distance is non-zero. Would be nice to have this fixed at some point.
